pkg/windows/windows.go

```diff
@@ -417,44 +413,30 @@ func (vm *windows) EnsureRequiredServicesStopped() error {
 	return nil
 }
 
-// ensureServicesAreRemoved ensures that all services installed by WMCO are removed from the instance
-func (vm *windows) ensureServicesAreRemoved() error {
-	// TODO: In this transitional period of migrating services to WICD's responsibility, WICD must be stopped first to
-	//       ensure it does not start any of these stopped services. This code should be removed as part of
-	//       https://issues.redhat.com/browse/WINC-733
-	svc := &service{name: wicdServiceName}
-	if err := vm.ensureServiceNotRunning(svc); err != nil {
-		return errors.Wrapf(err, "could not stop service %s", wicdServiceName)
+func (vm *windows) RunWICDCleanup(apiServerURL string, watchNamespace string, preserveNode bool) error {
+	wicdCleanupCmd := fmt.Sprintf("%s cleanup --api-server %s --sa-ca %s --sa-token %s --namespace %s",
+		wicdPath, apiServerURL, wicdCAFile, wicdTokenFile, watchNamespace)
+	if preserveNode {
+		wicdCleanupCmd = wicdCleanupCmd + " --preserveNode"
 	}
-	for _, svcName := range append(RequiredServices, AzureCloudNodeManagerServiceName) {
-		svc := &service{name: svcName}
-
-		// If the service is not installed, do nothing
-		exists, err := vm.serviceExists(svc.name)
-		if err != nil {
-			return errors.Wrapf(err, "unable to check if %s service exists", svc.name)
-		}
-		if !exists {
-			continue
-		}
-
-		// Make sure the service is stopped before we attempt to delete it
-		if err := vm.ensureServiceNotRunning(svc); err != nil {
-			return errors.Wrapf(err, "could not stop service %s", svc.name)
-		}
-		if err := vm.deleteService(svc); err != nil {
-			return errors.Wrapf(err, "could not delete service %s", svcName)
-		}
-		vm.log.Info("deconfigured", "service", svc.name)
+	if out, err := vm.Run(wicdCleanupCmd, true); err != nil {
+		vm.log.Info("failed to cleanup node", "command", wicdCleanupCmd, "output", out)
+		return err
 	}
 	return nil
 }
 
-func (vm *windows) Deconfigure() error {
+func (vm *windows) Deconfigure(preserveNode bool, watchNamespace string, apiServerURL string) error {
 	vm.log.Info("deconfiguring")
-	if err := vm.ensureServicesAreRemoved(); err != nil {
-		return errors.Wrap(err, "unable to remove Windows services")
+
+	if err := vm.deconfigureWICD(); err != nil {
+		return err
+	}
+	if err := vm.RunWICDCleanup(apiServerURL, watchNamespace, preserveNode); err != nil {
+		return errors.Wrap(err, "Unable to cleanup the Windows instance")
 	}
+	// wait for log files to be closed after cleanup
+	time.Sleep(time.Second * 30)
```

Comment:
This is risky. There is potential for `removeDirectories()` to still fail on busy instances where it could take more than 30 seconds for log files to be closed. Safer would be to retry `removeDirectories()` similar to `ensureHNSNetworksAreRemoved()`.

But what is still puzzling is why is the following not returning an error?
```
2023-01-12T16:25:28Z	DEBUG	wc 10.0.228.20	run	{"cmd": "cmd /c if exist C:\\var\\log\\ rmdir C:\\var\\log\\ /s /q", "out": "C:\\var\\log\\HYBRID~1\\hybrid-overlay.log - The process cannot access the file because it is being used by another process.\r\n"}
```
Is it because of the `cmd /c` prefix not propagating the error?

In addition, this is the general problem @sebsoto is also facing with hcs-shim binary.

Reply:
I missed:
```
2023-01-12T16:25:27Z	DEBUG	wc 10.0.228.20	run	{"cmd": "cmd /c if exist C:\\k\\ rmdir C:\\k\\ /s /q", "out": "C:\\k\\hybrid-overlay-node.exe - Access is denied.\r\n"}
```
This means that hybrid-overlay is still running. So even though the service is stopped and deleted the process is still running??? This is very strange.

Reply:
The sleep is working, it seems like it takes longer to close the log files as I mentioned. I can add the retry change as per your suggestion and see if it is helpful.

Reply:
PTAL at https://github.com/openshift/windows-machine-config-operator/pull/1372/commits/9a25d79b582df3395a093edd854dd3690125ebf9

Reply:
I cherry-picked @sebsoto 's commit and another [commit](https://github.com/openshift/windows-machine-config-operator/pull/1332/commits/5f3fae1a3aad3ddbda8c8db21921efefeb89dcc9) for a retry, hitting a connection reset now:
```
2023-01-17T10:53:43Z	ERROR	wc 10.0.251.204	error running	{"cmd": "cmd /c if exist C:\\var\\log\\kubelet\\ rmdir C:\\var\\log\\kubelet\\ /s /q", "out": "", "error": "read tcp 10.0.213.9:36156->10.0.251.204:22: read: connection reset by peer"}
github.com/openshift/windows-machine-config-operator/pkg/windows.(*windows).removeDirectories.func1
	/build/windows-machine-config-operator/pkg/windows/windows.go:584
```
Kicking off the reconciliation after node has been deleted is not useful, hence the retry is added.

Reply:
Please add a story to remove this sleep as we cannot cut a release with this.